kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/containExactlyInAnyOrder.kt

```diff
@@ -120,7 +120,9 @@ fun <T, C : Collection<T>> containExactlyInAnyOrder(
 
    val passed = expectedGroupedCounts.size == valueGroupedCounts.size
       && expectedGroupedCounts.all { (k, v) ->
-      valueGroupedCounts.filterKeys { verifier?.verify(k, it)?.areEqual() ?: (k == it) }[k] == v
+      // account for the case when a key might pass verifier equality but not default equality
+      val key = valueGroupedCounts.keys.find { verifier?.verify(k, it)?.areEqual() ?: (k == it) }
+      valueGroupedCounts.filterKeys { key == it }[key] == v
```

Comment:
If `actual` is `listOf("Apple", "apple")`, and value is `listOf("APPLE","APPLE")`, then `expectedGroupedCounts` is `mapOf("Apple" to 1, "apple" to 1)`, and `valueGroupedCounts` is `mpaOf("APPLE" to 2)`. If the verifier matches strings in case-insensitive way, this fix won't work - these two collections should match and they won't.

Reply:
the code to reproduce:
```
            val caseInsensitiveStringEquality: Equality<String> = object : Equality<String> {
               override fun name() = "Case Insensitive String Matcher"

               override fun verify(actual: String, expected: String): EqualityResult {
                  return if(actual.uppercase() == expected.uppercase())
                     EqualityResult.equal(actual, expected, this)
                  else
                     EqualityResult.notEqual(actual, expected, this)
               }
            }
            listOf("apple", "Apple") should containExactlyInAnyOrder(listOf("APPLE", "APPLE"), caseInsensitiveStringEquality)
```